packages/@aws-cdk/core/lib/private/runtime-info.ts

```diff
@@ -1,95 +1,81 @@
-import { basename, dirname } from 'path';
-import * as cxschema from '@aws-cdk/cloud-assembly-schema';
-import { major as nodeMajorVersion } from './node-version';
+import { IConstruct } from '../construct-compat';
+import { Stack } from '../stack';
+import { Stage } from '../stage';
 
-// list of NPM scopes included in version reporting e.g. @aws-cdk and @aws-solutions-konstruk
-const ALLOWED_SCOPES = ['@aws-cdk', '@aws-cdk-containers', '@aws-solutions-konstruk', '@aws-solutions-constructs', '@amzn'];
-// list of NPM packages included in version reporting
-const ALLOWED_PACKAGES = ['aws-rfdk', 'aws-cdk-lib', 'monocdk'];
+const ALLOWED_FQN_PREFIXES = [
+  // SCOPES
+  '@aws-cdk/', '@aws-cdk-containers/', '@aws-solutions-konstruk/', '@aws-solutions-constructs/', '@amzn/',
+  // PACKAGES
+  'aws-rfdk.', 'aws-cdk-lib.', 'monocdk.',
+];
 
 /**
- * Returns a list of loaded modules and their versions.
+ * Symbol for accessing jsii runtime information
+ *
+ * Introduced in jsii 1.19.0, cdk 1.90.0.
  */
-export function collectRuntimeInformation(): cxschema.RuntimeInfo {
-  const libraries: { [name: string]: string } = {};
-
-  for (const fileName of Object.keys(require.cache)) {
-    const pkg = findNpmPackage(fileName);
-    if (pkg && !pkg.private) {
-      libraries[pkg.name] = pkg.version;
-    }
-  }
+const JSII_RUNTIME_SYMBOL = Symbol.for('jsii.rtti');
 
-  // include only libraries that are in the allowlistLibraries list
-  for (const name of Object.keys(libraries)) {
-    let foundMatch = false;
-    for (const scope of ALLOWED_SCOPES) {
-      if (name.startsWith(`${scope}/`)) {
-        foundMatch = true;
-      }
-    }
-    foundMatch = foundMatch || ALLOWED_PACKAGES.includes(name);
+/**
+ * Source information on a construct (class fqn and version)
+ */
+export interface ConstructInfo {
+  readonly fqn: string;
+  readonly version: string;
+}
 
-    if (!foundMatch) {
-      delete libraries[name];
-    }
+export function constructInfoFromConstruct(construct: IConstruct): ConstructInfo | undefined {
+  const jsiiRuntimeInfo = Object.getPrototypeOf(construct).constructor[JSII_RUNTIME_SYMBOL];
+  if (typeof jsiiRuntimeInfo === 'object'
```

Comment:
I am wondering if we should be more defensive and throw if `JSII_RUNTIME_SYMBOL` is defined on the class but for some reason is malformed. If, for example, something is changed at the jsii level and we forget to update this code, we will lose lots of analytics information.